client/src/components/PresentedOptions.js

```diff
@@ -0,0 +1,504 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { inject } from 'lib/Injector';
+import {
+  Button, FormGroup, FormText, Input, InputGroup, InputGroupAddon, Label, Row, Col
+} from 'reactstrap';
+import fieldHolder from 'components/FieldHolder/FieldHolder';
+import CKANApi from 'lib/CKANApi';
+import i18n from 'i18n';
+
+/**
+ * "Presented options" are a either a selection of checkboxes, or a free text input field
+ * for the user to define a list of the presented options that will be applied for a given
+ * dropdown filter in the CMS.
+ *
+ * If using a checkbox set, the user can adjust the default delimiter field.
+ */
+class PresentedOptions extends Component {
+  constructor(props) {
+    super(props);
+
+    const value = props.value || {};
+    this.state = {
+      customOptions: [],
+      selectType: props.selectTypeDefault,
+      selections: {},
+      suggestedOptions: [],
+      suggestedOptionCache: {},
+      loading: false,
+      separatorDelimiter: '',
+      usedSeparators: [],
+      ...value,
+    };
+
+    this.handleCheckboxChange = this.handleCheckboxChange.bind(this);
+    this.handleInputChange = this.handleInputChange.bind(this);
+    this.handleSelectTypeChange = this.handleSelectTypeChange.bind(this);
+    this.handleDelimiterChange = this.handleDelimiterChange.bind(this);
+    this.handleExecuteSeparator = this.handleExecuteSeparator.bind(this);
+  }
+
+  componentDidMount() {
+    // Ensure the suggested options are loaded
+    this.loadSuggestedOptions();
+  }
+
+  componentDidUpdate(prevProps) {
+    // Compare if selected fields have changed
+    const newFields = this.props.selectedFields;
+    const oldFields = prevProps.selectedFields;
+
+    // If the type or length doesn't match then they must be different
+    if (typeof newFields !== typeof oldFields || newFields.length !== oldFields.length) {
+      this.loadSuggestedOptions();
+      return;
+    }
+
+    // Otherwise filter out options that appear in both arrays and assert the remaining count is
+    // zero
+    if (oldFields.filter(old => !newFields.includes(old)).length === 0) {
+      return;
+    }
+
+    this.loadSuggestedOptions();
+  }
+
+  /**
+   * Returns the namespaced form field name for the given field
+   *
+   * @param {string} fieldName
+   * @param {object} props If provided, will be used instead of this.props
```

Comment:
This was originally so you could pass props from the constructor before they're assigned to the instance - that code is gone now so maybe we can remove this extra argument?